apps/web/utils/calculations.ts

```diff
@@ -0,0 +1,89 @@
+// repo
+import {
+  VariableValueCalculationType,
+  VariableValueToSetCalculation,
+} from "@repo/data/useWalkthroughData";
+// NOTE: this feels weird, but it makes sure this module is something we can spy on for tests
+import * as ThisModule from "./calculations";
+
+export const calculateVariableValueToSet = (
+  answerValueToUse: number,
+  variableValueToSetCalculation: VariableValueToSetCalculation,
+): number => {
+  switch (variableValueToSetCalculation.variableValueCalculationType) {
+    case VariableValueCalculationType.Then:
+      if (variableValueToSetCalculation.variableValueToSetItemCalculations) {
+        // reduce with the answerValueToUse
+        return variableValueToSetCalculation.variableValueToSetItemCalculations.reduce<number>(
+          (acc, item) => {
+            return ThisModule.calculateVariableValueToSet(
+              acc,
+              item as VariableValueToSetCalculation,
+            );
+          },
+          answerValueToUse,
+        );
+      }
+
+      throw new Error(
+        "variableValueToSetItemCalculations must exist when variableValueCalculationType is Then",
+      );
```

Comment:
Same with these, are they handled or do they bomb the application if they occur?

Reply:
They're just father down the chain. They're caught! (thanks for asking these questions, good to verify!)